### PR TITLE
Convert  to Elmish

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

File: src/Scenes/CheckLocation.fs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+         ImagePickerOptions.TakePhotoButtonTitle "Kamera"
+         ImagePickerOptions.ChooseFromLibraryButtonTitle "Bibliothek"
+         AllowsEditing true]
+
+// Update
+let update msg model : Model*Cmd<AppMsg> =
+    match msg with
+    | LocationCheckMsg.SaveAndGoBack ->
+        match model.Status with
+        | Unchanged -> model,Cmd.ofMsg LocationCheckMsg.GoBack |> Cmd.map LocationCheckMsg
```
Comment (et1975) [inline src/Scenes/CheckLocation.fs]: just reutn the `GoBack` command, use `Cmd.map` in the parent.

File: src/Nightwatch.fs [hunk 2]
```diff
@@ -1,44 +1,37 @@
...
+        if !loaded then
+            Fable.Helpers.ReactNative.removeOnHardwareBackPressHandler backHandler
+
+        loaded := false
+
+    member x.render () =
+        if !loaded then
+            view this.state dispatch
+        else
+            view (init() |> fst) dispatch
```
Comment (et1975) [inline src/Nightwatch.fs]: shouldn't need to call the `init`, the program's `run` will do that.

File: src/Scenes/App.fs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+// App
+let program =
+    Program.mkProgram init update
+    |> Program.withConsoleTrace
+
+let loaded = ref false
+
+let view (model:AppModel) (dispatch: AppMsg -> unit) =
+    match model.SubModel with
+    | HomeModel model -> Home.view model dispatch
```
Comment (et1975) [inline src/Scenes/App.fs]: compose `dispatch` with the child's DU case to get keep the child unaware of the parent's type:
`HomeSceneMessage >> dispatch`

File: src/Scenes/CheckLocation.fs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                { model.LocationCheckRequest with 
+                      Status = newStatus
+                      Date = Some DateTime.Now }
+            Status = Changed }, []
+    | LocationCheckMsg.Error e ->
+        { model with
+            Status = Error (string e.Message) }, []
+
+// View
+let view (model:Model) (dispatch: AppMsg -> unit) =
```
Comment (et1975) [inline src/Scenes/CheckLocation.fs]: Take local message type in the dispatch signature, not parent's.

File: src/Scenes/App.fs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+| LocationListModel of LocationList.Model
+| CheckLocationModel of CheckLocation.Model
+
+type AppModel = {
+    SubModel : SubModel
+    NavigationStack: Page list
+}
+
+let wrap ctor model (subModel,cmd)  =
+    { model with SubModel = ctor subModel },cmd
```
Comment (et1975) [inline src/Scenes/App.fs]: This is where you want to map the child message type with `Cmd.map`

File: src/Scenes/App.fs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    | NavigateTo page -> navigateTo page (page::model.NavigationStack) model
+    | NavigateBack -> 
+        match model.NavigationStack with
+        | _::page::rest -> navigateTo page (page::rest) model
+        | _ -> model,Cmd.ofMsg ExitApp
+    | ExitApp -> 
+        Fable.Helpers.ReactNative.exitApp() 
+        model,Cmd.none
+
+let init() =
```
Comment (et1975) [inline src/Scenes/App.fs]: Consider composing init with the load from the db:
https://github.com/et1975/fable-elmish/blob/master/samples/react/todomvc/todomvc.fsx#L315

Comment (et1975) [inline src/Scenes/App.fs]: Another way to parametrize `init` would be the new `arg` type parameter introduced in elimsh 0.3.x - now you can use `Program.runWith` to supply something other than `unit` into the `init`.

File: src/Scenes/CheckLocation.fs (hunk 1 above)
Comment (forki) [inline src/Scenes/CheckLocation.fs]: how? that's a different type?!

File: src/Nightwatch.fs (hunk 2 above)
Comment (forki) [inline src/Nightwatch.fs]: It does. but this is needed for React native to recover after BackAndroid.exitApp

**Pull request merged**